### PR TITLE
[第3組]Refactor naming convetions

### PR DESCRIPTION
**Pull request by patrickhicetc**
### Purpose
Improve the variable names that are difficult to read, making the code more readable.

### Changes
* magic number
    * 2 -> news_channelId
    * 1 -> first_page
* global variables
    * bgs -> Scheduler
* local variables
    * get_new_info a -> paginated_news_data
    * get_new_info p -> pages
    * get_new_info p, p2 -> query_params
    * get_new_info l -> news_page
    * get_new m -> messagesToAi
    * get_new ai -> ai_response
    * get_new result -> completion_result
    * check_user_password_is_correct OuO -> user_record
    * get_article_upvote_details cnt -> upvote_count
    * read_news n -> news
    * read_news u -> user
    * read_user_news news -> news_articles
    * read_user_news result -> articles_with_upvotes
    * PromptRequest m -> messagesToAi
    * search_news m -> messagesToAi
    * search_news e -> error
    * news_summary m -> messagesToAi
    * news_summary result -> completion_result
* function
    * verify p1 -> plain_password
    * verify p2 -> hashed_password
    * check_user_password_is_correct n -> username
    * check_user_password_is_correct pwd -> password
    * get_article_upvote_details uid -> user_id
    * upvote_article id -> article_id
    * upvote_article u -> user
    * toggle_upvote n_id -> news_id
    * toggle_upvote u_id -> user_id
    * news_exists id2 -> article_id

### CODE REVIEW

**Comment by siangting** [comment]
The comment on the PR is excellent.

**Comment by Steven0811** [comment]
It is excellent that your refactored names make variables easier to read, though some magic numbers are still not extracted. Using "news_summary" as a function name is not good since it is not a verb or verb phrase. It can be changed to "summarize_news"

**Comment by WANG-YongLE** [comment]
優點：將 channelid 提取變數
建議:
將 291 行和 304 行的 token 提取變數。

**Comment by Rogue8464** [comment]
有把news_channelID跟first_page以及其他magic number提取出來，每個variable跟function都命名得很精確，但不應該把.scannerwork資料夾加進來

**Comment by Yi-Cheng-Wang** [comment]
**Evaluation:**
- Degree of variable refactoring is moderate
- Variable renaming is decent
- Code layout is tidy
- Commits are good
- Clarity of the PR is moderate

**Additional Evaluation:**
- The handling of magic numbers on lines 148–149 is very good; it is one of the few instances where this has been addressed!

**Suggestions:**
- There are still magic numbers that can be addressed; for example, the value `1892dhianiandowqd0n` on line 290 could be passed in a better way.
- The variable `expires_delta` on line 295 is not easily understandable and should be changed.
- For consistency, the multiple instances of `messagesToAi` on line 196 and others should be changed to `messages_to_AI`.

**Comment by mlcp29th-hi** [comment]
Hello,
1. 依照 PEP 8 Style Guideline，變數命名應該要用 `snake_case` 命名如 `scheduler`、`background_scheduler` 等等，否則以下程式碼：
```py
Scheduler.add_job()
Scheduler.start()
Scheduler.stop()
```
容易被解讀成有個 class 叫做 `Scheduler`，然後呼叫其中的 `class method`（隸屬於 class 本身而不是其 instance 的成員）。

2. 如上，`messagesToAi` 建議改成 `messages_to_ai` 以符合 PEP 8 並維持程式在此專案中的一致性。

3. 建議命名時可以盡量避免諸如 data/info/result 等形式，否則以旁人的角度較不容易立刻理解 `news`、`new_data`、`news_info`、`paginated_news_data `、`completion_result`等的意義／差別，畢竟變數本身即資料，`_data` 等後綴對於理解其意義的幫助有限。

4. `verify` 不僅單純執行動作，還會回傳 boolean 表示密碼正確與否，這樣的函式命名時可以考慮以 `is_`、`has_`、`does_` 開頭，在語意上會更流暢如 `is_password_correct()`。

5. `delete_stmt`、`insert_stmt` 可以考慮改成 `delete_statement`、`insert_statement` 使人們討論此程式時便於發音，較利於溝通。

6. 若是要將 `news_channel_id`、`first_page` 等常數提取出來，建議依照 PEP 8 以 `UPPER_CASE` 的形式命名，如 `NEWS_CHANNEL_ID = 2`、`FIRST_PAGE = 1`。不過個人覺得此處沒有提取 magic number 的必要：
```py
query_params = {
    "page": first_page,
    "id": f"search:{quote(search_term)}",
    "channelID": news_channelID,
    "type": "search word"
}
```
因為根據其結構我們已經知道 1 是 page 欄位的參數、2 是 channelID 欄位的參數。額外再將此他們獨立成變數 `FIRST_PAGE`、`NEWS_CHANNEL_ID` 並沒有提供更詳細的上下文，反而會產生冗餘的程式。

7.  基於一個 Pull Request 應該只專注於一個目的的原則，與 SonarQube 相關的更動如 .scannerwork 資料夾和 docker-compose.yml 另外獨立成新的 Pull Request 會更好。

File: backend/main.py [hunk 1]
```diff
@@ -145,7 +145,8 @@ def add_new(news_data):
     session.commit()
     session.close()
 
-
+news_channelId = 2
+first_page = 1
```
Comment (taro0513) [inline backend/main.py:149]: The magic number is usually not the dynamic variable. According to PEP 8, the constant variable should defined on a module level and written in all capital letters with underscores separating words

[reference](https://peps.python.org/pep-0008/#constants)

**Review by taro0513 — COMMENTED** [review]
If the commit contains the change accidentally, we could rebase or revert the commit according to the stage at which you find the mistake

**Pull request merged**